workflow_tests/assertions/preDeployAssertions.js

```diff
@@ -0,0 +1,253 @@
+const utils = require('../utils/utils');
+
+const assertLintJobExecuted = (workflowResult, didExecute = true) => {
+    const steps = [utils.createStepAssertion('Run lint workflow', true, null, 'LINT', 'Running lint workflow')];
+
+    steps.forEach((expectedStep) => {
+        if (didExecute) {
+            expect(workflowResult).toEqual(expect.arrayContaining([expectedStep]));
+        } else {
+            expect(workflowResult).not.toEqual(expect.arrayContaining([expectedStep]));
+        }
+    });
+};
+
+const assertTestJobExecuted = (workflowResult, didExecute = true) => {
+    const steps = [utils.createStepAssertion('Run test workflow', true, null, 'TEST', 'Running test workflow')];
+
+    steps.forEach((expectedStep) => {
+        if (didExecute) {
+            expect(workflowResult).toEqual(expect.arrayContaining([expectedStep]));
+        } else {
+            expect(workflowResult).not.toEqual(expect.arrayContaining([expectedStep]));
+        }
+    });
+};
+
+const assertIsExpensifyEmployeeJobExecuted = (workflowResult, didExecute = true) => {
+    const steps = [
+        utils.createStepAssertion('Get merged pull request', true, null, 'IS_EXPENSIFY_EMPLOYEE', 'Getting merged pull request', [{key: 'github_token', value: '***'}]),
+        utils.createStepAssertion('Check whether the actor is member of Expensify/expensify team', true, null, 'IS_EXPENSIFY_EMPLOYEE', 'Checking actors Expensify membership', [
+            {key: 'GITHUB_TOKEN', value: '***'},
+            {key: 'username', value: 'Dummy Author'},
+            {key: 'team', value: 'Expensify/expensify'},
+        ]),
+    ];
+
+    steps.forEach((expectedStep) => {
+        if (didExecute) {
+            expect(workflowResult).toEqual(expect.arrayContaining([expectedStep]));
+        } else {
+            expect(workflowResult).not.toEqual(expect.arrayContaining([expectedStep]));
+        }
+    });
+};
+
+const assertNewContributorWelcomeMessageJobExecuted = (workflowResult, didExecute = true, isOsBotify = false, isFirstPr = false) => {
+    const steps = [
+        utils.createStepAssertion('Checkout', true, null, 'NEW_CONTRIBUTOR_WELCOME_MESSAGE', 'Checking out', [{key: 'token', value: '***'}]),
+        utils.createStepAssertion('Get merged pull request', true, null, 'NEW_CONTRIBUTOR_WELCOME_MESSAGE', 'Getting merged pull request', [{key: 'github_token', value: '***'}]),
+        utils.createStepAssertion(isOsBotify ? 'Get PR count for OSBotify' : 'Get PR count for Dummy Author', true, null, 'NEW_CONTRIBUTOR_WELCOME_MESSAGE', 'Getting PR count', [
+            {key: 'GITHUB_TOKEN', value: '***'},
+        ]),
+    ];
+    const osBotifyBody =
```

Comment:
NAB - is there a way to get these from somewhere else, so that these tests won't break if we end up changing the copy?

Reply:
New assertion files can be generated using `npm run workflow-test:generate` command. So if at any point in time you change the workflow file and the assertions become outdated you can regenerate a new one based on the updated workflow with this command. It will however remove any changes added to the assertions file manually, so it would probably be best to keep a copy of the old file, generate new one and cherry pick what fits from both.
Or just copy the text from the new workflow